smallclib/Makefile

```diff
@@ -0,0 +1,140 @@
+P	= mips-mti-elf-
+CC	= $(P)gcc -mips32r2 -EL -nostdlib
+AR  = $(P)ar
+
+# -Werror disabled for building a third-party external lib
+CFLAGS =  -Wall -DPIC32MZ
+# No optimisations, emit debugging info.
+CFLAGS += -O0 -g
```

Comment:
On 18 December 2015 at 13:43, Krystian Bacławski <notifications@github.com>
wrote:

> In smallclib/Makefile
> <https://github.com/cahirwpz/wifire-os/pull/11#discussion_r48021339>:
>
> > @@ -0,0 +1,140 @@
> > +P	= mips-mti-elf-
> > +CC	= $(P)gcc -mips32r2 -EL -nostdlib
> > +AR  = $(P)ar
> > +
> > +# -Werror disabled for building a third-party external lib
> > +CFLAGS =  -Wall -DPIC32MZ
> > +# No optimisations, emit debugging info.
> > +CFLAGS += -O0 -g
>
> Why not just -Og i.e. optimizations that do not interfere with generating
> debugging information ?
>
I'm afraid that gcc uses quite unusual definition of "not interfering with
debugging" and even with -Og you can still sometimes see that annoying
"<optimized out>" in gdb.

However, it may be a good idea to have both debug (-O0) and release (-O3
-flto) builds since optimizations may help finding some bugs in the code
(mostly undefined behaviours, sadly there is no easy way to get ubsan
working for a kernel) and lto has problems of it own, in most cases fixed
by adding some [[gnu::used]] attributes (on the other hand lto can quite
significantly reduce the size of the binary which may be quite important
for this project).